### PR TITLE
AB#8504,8505,8541 - Refactor Create/Edit event flow, Refactor Review/Details pages

### PR DESCRIPTION
**Pull request by sc-antonisnikolaidis**
Things changed in this PR:

- Extract Event Detail into a separate component to be reused among separated Review/Details pages.
- Extract FieldsEvent into a separate component to be shared between Create/Edit pages.
- Control required/single fields from FIELD_OVERRIDES constant (hard-coded by us but in one place for all forms).
- Refactor DatePicker component.
- Add InputBoolean for boolean fields.
- Change 3 step creation to 2 step creation (overview + detailed view).
- Block going back on the Overview page.
- Allow Discard from the Create Details page.
- Disable submit for all required fields in FIELD_OVERRIDES.
- Add Discard Changes screen controlled by route params.
- Refactor state to have event, athlete specific keys removed and replaced with dynamic UUIDs (allows nested flows like creating a new event from an existing new event flow).

### CODE REVIEW

File: play-media-native/helpers/events.ts [hunk 1]
```diff
@@ -57,3 +59,19 @@ export const prepareRequestFields = (event: Event) => {
...
+
+export const canSubmitEvent = (fields: IIndexable) => {
+  if (!fields) {
+    return false;
+  }
+
+  let canSubmit = true;
+
+  for (const field in EVENT_FIELD_OVERRIDES) {
+    if (EVENT_FIELD_OVERRIDES[field].required && !fields[field]) {
```
Comment (sc-antonisnikolaidis) [inline play-media-native/helpers/events.ts:71]: needs to change

File: play-media-native/interfaces/indexable/index.ts [hunk 2]
```diff
@@ -1,3 +1,15 @@
+import { ReactNode } from 'react';
+
 export interface IIndexable {
   [key: string]: any;
 }
+
+export interface IFieldOverride {
```
Comment (sc-antonisnikolaidis) [inline play-media-native/interfaces/indexable/index.ts:7]: move somewhere else

File: play-media-native/features/FieldsEvent/FieldsEvent.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import { CONTENT_TYPES, FIELD_TYPES } from '../../constants/contentTypes';
+import { ContentItemFields } from '../ContentItemFields/ContentItemFields';
+
+const FIELD_OVERRIDES = {
```
Comment (sc-antonisnikolaidis) [inline play-media-native/features/FieldsEvent/FieldsEvent.tsx:4]: move to constants

File: play-media-native/components/InputBoolean/InputBoolean.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        onValueChange={onChange}
+        thumbColor={theme.colors.white.DEFAULT}
+        trackColor={{ false: theme.colors.gray.DEFAULT, true: theme.colors.green.DEFAULT }}
+        value={inputValue}
+      />
+    );
+  }, [inputValue, onChange]);
+
+  if (!title) {
+    return <>{switchRenderer}</>;
```
Comment (sc-antonisnikolaidis) [inline play-media-native/components/InputBoolean/InputBoolean.tsx:37]: needs required star even if no title

File: play-media-native/components/DatePicker/DatePicker.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ export const DatePicker = ({ value, visible, onChange, onClose }) => {
           display="inline"
           themeVariant="dark"
           accentColor={theme.colors.yellow.DEFAULT}
-          value={new Date(value)}
+          value={value ? new Date(value) : new Date()}
```
Comment (sc-antonisnikolaidis) [inline play-media-native/components/DatePicker/DatePicker.tsx:23]: create value var above

File: play-media-native/constants/event.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  teaser: '',
+  timeAndDate: null,
+  sport: null,
+  featuredImage: null,
+  relatedMedia: [],
+  athletes: [],
+  similarEvents: [],
+};
+
+export const EVENT_FIELD_OVERRIDES: IIndexable = {
```
Comment (sc-antonisnikolaidis) [inline play-media-native/constants/event.ts:17]: maybe not needed anymore

File: play-media-native/features/CardSport/CardSport.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        marginBottom: theme.spacing.xs,
+        ...style,
+      }}
+    >
+      <Card.Cover
+        style={{
+          backgroundColor: theme.colors.black.DEFAULT,
+          height: 140,
+        }}
+        source={{ uri: null }}
```
Comment (sc-antonisnikolaidis) [inline play-media-native/features/CardSport/CardSport.tsx:74]: fix this

File: play-media-native/screens/EditEvent.tsx [hunk 8]
```diff
@@ -1,121 +1,50 @@
...
+export const EditEventScreen = ({ navigation, route }) => {
+  const [stateKey] = useState(route?.params?.stateKey);
+  const { contentItems, editMultiple, reset } = useContentItems();
+
+  const event = useMemo(
+    () => contentItems[stateKey] ?? null,
+    [contentItems, stateKey]
+  ) as unknown as Event;
+
+  const [fields, setFields] = useState<IIndexable>({
```
Comment (sc-antonisnikolaidis) [inline play-media-native/screens/EditEvent.tsx:22]: needs cleanup here

File: play-media-native/screens/AddSport.tsx [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+  if (single) {
+    return (
+      <Screen>
+        <Listing
+          data={sports}
+          isLoading={isFetchingInitialSports}
+          numColumns={2}
+          renderItem={({ item }) => (
+            <CardSport
```
Comment (sc-nelikostadinova) [inline play-media-native/screens/AddSport.tsx:100]: There is something wrong with the Listing. Depending on what I have selected different sports appear. 
[Nothing selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334926423_197827312856969_2148915917303426376_n.jpg?_nc_cat=103&ccb=1-7&_nc_sid=ae9488&_nc_ohc=193dvMNuj6oAX_IJxA1&_nc_ht=scontent.fsof8-1.fna&oh=03_AdRPWPUECeYt8DKWsPyAOHJBL2b-EyZfYntNEZaO5uf8rA&oe=642D23D7)
[Skateboarding selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334927897_442254234733708_157601388328995673_n.jpg?_nc_cat=101&ccb=1-7&_nc_sid=ae9488&_nc_ohc=NvSkqMciR1kAX9PfKYD&_nc_ht=scontent.fsof8-1.fna&oh=03_AdS_DGBNZ5WGrdY465tcQhUsyJ3X9WQ_icBrPbab7gWzGA&oe=642CF43C)
[Climbing selected](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334880890_737352741131892_7410477830176955600_n.jpg?_nc_cat=102&ccb=1-7&_nc_sid=ae9488&_nc_ohc=Q3fciMf4QQsAX-PxK1H&_nc_ht=scontent.fsof8-1.fna&oh=03_AdQZsHRrcVCMdFodBmZSQkcDQSE6LpllcHrm9k8ClqvN1w&oe=642D0638)

When dragging down to refresh the listing gets cut off (not present in event and athlete listing screens).
[Refreshing](https://scontent.fsof8-1.fna.fbcdn.net/v/t1.15752-9/334901960_560136875902000_28462774421278257_n.jpg?_nc_cat=102&ccb=1-7&_nc_sid=ae9488&_nc_ohc=BOsP9Lq4UIMAX_sv4Ax&_nc_ht=scontent.fsof8-1.fna&oh=03_AdTufE5dCGhgJIBHmqN6gh6R0tDKeXYMW1tPixAuyeAkbA&oe=642D1DDE)

Also the numColumns don't seem to be working properly.

**Review by sc-nelikostadinova — CHANGES_REQUESTED** [review]

File: play-media-native/screens/AddSport.tsx (hunk 9 above)
Comment (sc-antonisnikolaidis) [reply play-media-native/screens/AddSport.tsx:100]: I pushed a fix that could possibly fix things on iOS, but I can't reproduce these on Android. This is what I see.

Selected sport
sport-selected

None selected
none-selected

Refreshing
refreshing

File: play-media-native/screens/AddSport.tsx [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    );
+  }
+
+  return (
+    <Screen>
+      <Listing
+        data={sports}
+        isLoading={isFetchingInitialSports}
+        numColumns={2}
+        renderItem={({ item }) => (
```
Comment (sc-nelikostadinova) [inline play-media-native/screens/AddSport.tsx:122]: A suggestion - should we extract the renderItem and have it be conditional based on the "single" variable. This way we will have only one place to make changes to the listing properties. And if we do the same for the BottomActions (have them extracted in a const that returns conditionally) we can have only one return statement.

Comment (sc-antonisnikolaidis) [reply play-media-native/screens/AddSport.tsx]: Fixed. Can you give it a quick test please cause we had some conflicts here?

**Review by sc-nelikostadinova — APPROVED** [review]

**Pull request merged**